src/tree/ingest.rs

```diff
@@ -28,6 +32,12 @@ impl<'a> Ingestion<'a> {
     ///
     /// Will return `Err` if an IO error occurs.
     pub fn new(tree: &'a Tree) -> crate::Result<Self> {
+        // Use the shared flush helper so ingestion participates in the same
+        // path as normal writes: any dirty memtable content is moved into
+        // tables before building new tables from the ingestion stream.
+        // This keeps the lookup path ordered as active > sealed > tables.
+        tree.flush_active_memtable(SeqNo::MAX)?;
```

Comment:
```rs
#[test]
fn ingestion_dirty_snapshot() -> lsm_tree::Result<()> {
    let folder = tempfile::tempdir()?;

    let seqno = SequenceNumberCounter::default();
    let tree = Config::new(&folder, seqno.clone()).open()?;

    tree.insert("a", "a", seqno.next());
    tree.insert("a", "b", seqno.next());

    let snapshot_seqno = 1;
    assert_eq!(b"a", &*tree.get("a", snapshot_seqno)?.unwrap());

    let mut ingest = tree.ingestion()?.with_seqno(seqno.next());
    ingest.write("b".into(), "b".into())?;
    ingest.finish()?;

    assert_eq!(b"a", &*tree.get("a", snapshot_seqno)?.unwrap());

    Ok(())
}
```

Using `SeqNo::MAX` breaks snapshot reads.

The more secure solution here would be to just use watermark = 0.